libraries/chain/db_market.cpp

```diff
@@ -406,6 +486,14 @@ bool database::fill_order( const limit_order_object& order, const asset& pays, c
       } );
    }
 
+   if( order.deferred_paid_fee.amount > 0 ) // implies head_block_time() > HARDFORK_CORE_604_TIME
+   {
+      const auto& fee_asset_dyn_data = order.deferred_paid_fee.asset_id(*this).dynamic_asset_data_id(*this);
+      modify( fee_asset_dyn_data, [&](asset_dynamic_data_object& addo) {
+         addo.accumulated_fees += order.deferred_paid_fee.amount;
```

Comment:
Hm, delaying the update of accumulated_fees until this point opens up an attack vector to keep the fee pool drained at virtually no cost. An attacker can create a limit order with a high USD fee that drains the pool of USD. He needs to lock up his USD for this, but he never has to actually pay the USD (as long as the order isn't filled of course).

Reply:
This is intended behavior.
When the order is cancelled, both he and the pool will be refunded.
When the order is filled or partially filled, he will pay the USD and pool will pay the CORE.
Worst case is he keeps the order open forever, then the funds will be locked forever.

New features to address this could be:
* force a delayed release: reduce maximum order expiration, so the funds will be released sooner, or
* force a delayed pay: if an order is not cancelled after a period, don't refund the fee.

However, an attacker can get around both of these with a bot automatically renewing the order.

Perhaps it's better to discuss this in bsips repository.